lib/bitvec.cpp

```diff
@@ -145,3 +146,20 @@ bool bitvec::is_contiguous() const {
         return false;
     return max().index() - min().index() + 1 == popcount();
 }
+
+/**
+ * The purpose of this function is to barrel_shift a bitvec around a pivot.  This function
+ * will crash if the pivot is smaller than the size of the bitvector
+ */
+bitvec bitvec::barrel_shift_left(size_t shift, size_t pivot) const {
+    if (max().index() > pivot) {
+        BUG("Barrel shifting around pivot %d is smaller than the max of the bitvec %d", pivot,
+            max().index());
```

Comment:
This shouldn't be a problem -- the below code should work fine for this, rotating 0 bits from above max().index() just fine.